File: 400/492 - Construct the Rectangle.java
class Solution {
    public int[] constructRectangle(int area) {
        int w, l;

        for (l = (int)Math.sqrt(area); l <= area; ++l) {
            if (area % l != 0)
                continue;
            w = area / l;
            if (w > l)
                continue;
            return new int[]{l, w};
        }
        
        return null;
    }
}

File: csharp/492.construct-the-rectangle.cs
/*
 * @lc app=leetcode id=492 lang=csharp
 *
 * [492] Construct the Rectangle
 */

using System;

// @lc code=start
public class Solution {
    public int[] ConstructRectangle(int area) {
        int w = (int) Math.Sqrt(area);
        while (true) {
            if (area % w == 0) {
                return new int[] { area / w, w };
            }
            --w;
        }
    }
}
// @lc code=end

